draft-rescorla-tls-esni.md

```diff
@@ -378,21 +411,25 @@ for other purposes within the same message (e.g., encrypting other
 values). Those usages MUST have their own HKDF labels to avoid
 reuse.
 
-[[OPEN ISSUE: If in future you were to reuse these keys for
+[[OPEN ISSUE: If in the future you were to reuse these keys for
 0-RTT priming, then you would have to worry about potentially
 expanding twice of Z_extracted. We should think about how
-to harmonize these to make sure that we maintain key separation.
-Similarly, if the server uses the same key for ESNI as it does
-in ServerKeyShare, this is going to involve re-use of Z in some
-hard to analyze ways. Of course, this would also involve
-abandoning PFS.]]
+to harmonize these to make sure that we maintain key separation.]]
 
 This value is placed in an "encrypted_server_name" extension.
 
 The client MAY either omit the "server_name" extension or provide
 an innocuous dummy one (this is required for technical conformance
 with {{!RFC7540}}; Section 9.2.)
 
+If the server does not provide an "encrypted_server_name" extension
+in EncryptedExtensions, the client MUST abort the connection with 
+an "illegal_parameter" alert. Moreover, it MUST check that 
+PaddedServerNameList.nonce matches the value of the 
+"encrypted_server_name" extension provided by the server. If it 
+does not, the client MUST abort the connection with an "illegal_parameter" 
+alert.
```

Comment:
This could be read as saying that the client must abort if it doesn't check. Maybe "provided by the server, and otherwise abort the connection ...."

Reply:
Yep, good suggestion. Fixed.